### PR TITLE
layer.conf: add 'scarthgap' to LAYERSERIES_COMPAT

### PR DESCRIPTION
**Pull request by hnez**
Recently CI jobs have started [failing](https://github.com/labgrid-project/meta-labgrid/actions/runs/8570392864/job/23488216369?pr=49) since poky master has changed its LAYERSERIES_COMPAT to scarthgap.

Add scarthgap compatibility and while at it remove mickledore, which has reach its EOL.

### CODE REVIEW

**Review by ejoerns — APPROVED** [review]

**Comment by hnez** [comment]
Hi @ejoerns,

it looks like the GitHub action ran out of disk space (at least according to bitbakes disk monitor).
Can this be fixed by providing appropriate sstate via the sstate mirror so that less compilation takes place in the action?
Or should we change the `BB_DISKMON_DIRS` to reduce the limit to below 1G?
Or move the `rust-llvm-native` build into its own bitbake call before the `autobahn` build so that its rm_work is executed before the python builds start?

**Comment by ejoerns** [comment]
> Hi @ejoerns,
> 
> it looks like the GitHub action ran out of disk space (at least according to bitbakes disk monitor). Can this be fixed by providing appropriate sstate via the sstate mirror so that less compilation takes place in the action?

Yes. I have re-triggered my failed sstate server job. Let's see how far it comes..

> Or should we change the `BB_DISKMON_DIRS` to reduce the limit to below 1G?

I wouldn't assume that this makes a notable difference

> Or move the `rust-llvm-native` build into its own bitbake call before the `autobahn` build so that its rm_work is executed before the python builds start?

Could work, but let's try updating/fixing the mirror first.

**Comment by ejoerns** [comment]
I see scarthgap sstate doesn't work anyway since the sstate job fails due to layer incompatibility. I'd tend to say we just merge this and check for possible fallout later.

**Comment by hnez** [comment]
I would have preferred to see the job succeed before merging, but would be okay with merging it as-is.
I do not have permissions to merge this so if anybody else wants to do the honors feel free to do so.

**Comment by Emantor** [comment]
I guess the correct youth term here is "YOLO".

**Pull request merged**
